light/store/db/db.go

```diff
@@ -243,26 +243,20 @@ func (s *dbs) Prune(size uint16) error {
 	b := s.db.NewBatch()
 	defer b.Close()
 
-	var pruned uint16 = 0
 	for itr.Valid() && numToPrune > 0 {
 		if err = b.Delete(itr.Key()); err != nil {
 			return err
 		}
 		itr.Next()
 		numToPrune--
-		pruned++
 	}
 	if err = itr.Error(); err != nil {
 		return err
 	}
 
 	// 3) // update size
-	s.mtx.Lock()
-	s.size -= pruned
-	newSize := s.size
-	s.mtx.Unlock()
-
-	if err := b.Set(s.sizeKey(), marshalSize(newSize)); err != nil {
+	s.size = size
```

Comment:
Is this the correct value? Before we were deducting from `s.size`.

Reply:
yup before we were deducting the pruned value from s.size but because this now has essentially a write lock on the database and the size number is not going to change in between reading and writing it we know what the size of the db is going to be after the prune operation and can assign it instead.